pkg/apis/pipeline/v1alpha2/pipeline_validation.go

```diff
@@ -194,6 +194,10 @@ func (ps *PipelineSpec) Validate(ctx context.Context) *apis.FieldError {
 		return err
 	}
 
+	// Validate PipelineTask runOn and runAfter
+	if err := validatePipelineTaskRunOn(ps.Tasks); err != nil {
+		return err
+	}
```

Comment:
validating `runOn` for not having any state other than whats allowed and making sure doesnt have duplicates